### PR TITLE
[lifecycle] Manage Un/Deployment of Meshery Adapters

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

- [x] Kubernetes
    - [x] Able to deploy/undeploy
   - [x] Remove OSM Adapter sub chart. (missed out in https://github.com/meshery/meshery/pull/8375)
- [x] Docker
  - [x] Able to deploy/undeploy
  - [x] Detect running Meshery container and install adapters' containers in same network

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by suhail34** [comment]
Doubt :
As you have removed adapters from docker-compose how will this thing work with mesheryctl as it uses the same docker-compose file and downloads it

**Comment by theBeginner86** [comment]
> Doubt :
> As you have removed adapters from docker-compose how will this thing work with mesheryctl as it uses the same docker-compose file and downloads it

You can make use of Adapter Tracker Interface. It does have the info for adapters as it's seeded on Meshery boot as env vars.
FYI: https://github.com/meshery/meshery/blob/master/server/cmd/main.go#L125

**Comment by theBeginner86** [comment]
This PR is ready.

System Flows:

1. For Docker env, performing un/deploying would create new containers (or stop and delete running containers) of each adapter. Would ensure that the new adapters are on the same Docker network as the one for Meshery container
2. For K8s env, performing un/deploying would internally use Helm with overrides for each adapter (enabled: true/false). Then on each un/deployment Helm upgrade is performed.

**Comment by theBeginner86** [comment]
_The initial code written for this functionality is not clean would require some refactoring considering most of this functionality can be extracted into MeshKit. Like provisioning of containers or supporting multi-k8s deployment through our deployment modals._

File: install/docker/docker-compose.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: "3"
 services:
   meshery:
-    image: layer5/meshery:latest
+    image: layer5/meshery:stable-latest
```
Comment (theBeginner86) [inline install/docker/docker-compose.yaml:4]: Looks like someone had changed this tag for meshery image. 🤔

As there is no image tag of latest for layer5/meshery image.  https://hub.docker.com/r/layer5/meshery/tags?page=1&name=latest

**Comment by theBeginner86** [comment]
I have one more question,
Should we start publishing all Meshery Adapter's charts separately too?

So that users can directly do following for managing deployments of their individual adapters irrespective of Meshery deployment:
`helm install meshery/meshery-istio`

Instead of deploying them as follows:
```
helm pull meshery/meshery --untar
cd meshery/charts
helm install meshery-istio ./meshery-istio
```


The above would have no change in the current sub chart structure that we have its just that instead of referring a local charts dir we would be referring the remotely published charts.

The above would also be helpful when we would be deploying Meshery in K8s and trying to manage lifecycle of Adapters. As currently, since both Meshery and Adapters are part of the same release so during deployment/un-deployment of any adapter, the Meshery release is upgraded that leads to restart of Meshery pods. So, if Adapters charts are published separately then we can easily put them under separate release and that would not hinder Meshery pods.

**Comment by leecalcote** [comment]
I'm not opposed to it, @theBeginner86.

@AugustasV @nebula-aac @Revolyssup thoughts?

**Comment by leecalcote** [comment]
Good error messages.

**Comment by leecalcote** [comment]
Error handling of local builds that try to perform deployments works.

**Pull request merged**